### PR TITLE
CLI init check

### PR DESCRIPTION
**Pull request by sekulicd**
Prevent execution without initialisation by checking cfgData and returning error if not initialised.

@tiero please review.

### CODE REVIEW

**Comment by altafan** [comment]
@sekulicd is this still needed after the latest commits pushed on master? can you double check please?

**Comment by sekulicd** [comment]
> @sekulicd is this still needed after the latest commits pushed on master? can you double check please?

Before this, if user runs any cmd code would panic as config data is nil, so this checks if config data is not nil before each cmd except init, so yes this is still needed.

**Comment by tiero** [comment]
Can you merge master? we should already have the app.Version fix there

**Comment by sekulicd** [comment]
> Can you merge master? we should already have the app.Version fix there

Done

File: client/main.go [hunk 1]
```diff
@@ -388,6 +384,12 @@ func getArkSdkClient(ctx *cli.Context) (arksdk.ArkClient, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	commandName := ctx.Args().First()
```
Comment (tiero) [inline client/main.go:388]: Doing the check here, will then result in concatenating to other err messages ie. `error: error initializing ark sdk client: ` Is this what we really want? bit ugly. Is it possible to make this check more before others?

```
ark config
error: error initializing ark sdk client: CLI not initialized, run 'init' cmd to initialize
```

Comment (sekulicd) [reply client/main.go:388]: maybe we just fmt.Print and return without err?

File: client/main.go [hunk 2]
```diff
@@ -388,6 +384,12 @@ func getArkSdkClient(ctx *cli.Context) (arksdk.ArkClient, error) {
...
+	if commandName != "init" && cfgData == nil {
+		return nil, errors.New("CLI not initialized, run 'init' cmd to initialize")
```
Comment (altafan) [inline client/main.go]:
```suggestion
		return nil, fmt.Errorf("CLI not initialized, run 'init' cmd to initialize")
```

**Pull request merged**